sources/Debate/src/main/java/com/example/Debate/service/DebateServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.example.Debate.service;
+
+import com.example.Debate.dto.DebateDto;
+import com.example.Debate.model.Debate;
+import com.example.Debate.repository.DebateRepository;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class DebateServiceImpl implements DebateService{
+    private DebateRepository debateRepository;
+    private ModelMapper modelMapper;
+
+    public DebateServiceImpl(DebateRepository debateRepository, ModelMapper modelMapper) {
+        this.debateRepository = debateRepository;
+        this.modelMapper = modelMapper;
+    }
+
+    @Override
+    public DebateDto getDebateById(String id) {
+        Optional<Debate> debateOptional = debateRepository.findById(id);
+        return debateOptional.isEmpty() ? null : modelMapper.map(debateOptional.get(),DebateDto.class);
```

Comment:
returning nulls - an exception could have been thrown instead. we will add error handling in the upcoming sprints.